torch/csrc/dynamo/init.cpp

```diff
@@ -36,6 +36,11 @@ using torch::dynamo::autograd::torch_c_dynamo_compiled_autograd_init;
 
 namespace {
 
+std::map<std::pair<PyTypeObject*, PyTypeObject*>, bool> _isinstance_cache;
+std::vector<PyObject*> _all_subclasses;
+PyTypeObject* VT_tp = nullptr;
+PyTypeObject* LazyVT_tp = nullptr;
```

Comment:
Are these owned pointers?  My main worry is we get the recounting right.  Could we use something like `THPObjectPtr` to ensure we don't have pointers to dead objects?